jerry-core/jerryscript.h

```diff
@@ -402,6 +422,11 @@ jerry_value_t jerry_exec_snapshot (const uint32_t *snapshot_p, size_t snapshot_s
 size_t jerry_parse_and_save_literals (const jerry_char_t *source_p, size_t source_size, bool is_strict,
                                       uint32_t *buffer_p, size_t buffer_size, bool is_c_format);
 
+/*
+ * External context functions.
+ */
+jerry_instance_t *jerry_create_instance (uint32_t heap_size, jerry_instance_alloc_t alloc);
```

Comment:
If we don't pass the alloc function, then we may need 2 api to replace `jerry_create_instance`
``` c
jerry_get_instance_size(size_t heap_size);
jerry_init_instance(jerry_instance_t *instance_p, size_t heap_size)
```
In this way, users should first get the size, then alloc the instance on their own, then init the instance. And users will not know the internal details of the jerry_instance.

What do you think? @martijnthe @zherczeg